### PR TITLE
fix(cache): can not mkdir multilevel folder

### PR DESCRIPTION
**Pull request by iamxiaojianzheng**
about this https://github.com/oysandvik94/curl.nvim/issues/44

### CODE REVIEW

File: lua/curl/cache.lua [hunk 1]
```diff
@@ -21,7 +22,10 @@ local curl_cache_dir = function(custom_dir)
 		cache_dir = cache_dir / custom_dir ---@type Path
 	end
 
-	cache_dir:mkdir({ parents = true, exists_ok = true })
+	if vim.fn.mkdir(cache_dir:absolute(), "p") ~= 1 then
+		notify.error("create directory error: " .. cache_dir:absolute())
```
Comment (oysandvik94) [inline lua/curl/cache.lua:26]: This message goes to the user, we should perhaps formalize it a little bit:

"Error creating collection: could not create directory " .. cache_dir:absolute()

**Review by oysandvik94 — CHANGES_REQUESTED** [review]

**Comment by oysandvik94** [comment]
@iamxiaojianzheng Didn't realize that it would rewrite so much, you might need to recreate this PR :P

**Comment by iamxiaojianzheng** [comment]
> @iamxiaojianzheng Didn't realize that it would rewrite so much, you might need to recreate this PR :P

I don't know how you modify it, I just modify the information of a commit individually through lazygit. Other commits are not affected.

**Comment by oysandvik94** [comment]
@iamxiaojianzheng I used this: https://github.com/newren/git-filter-repo

I think is rewrites the history completely, so that the email is not in the history at all. 

Sorry for the complications, but maybe you need to etither sync your branch with main, or recreate your fork.

**Comment by iamxiaojianzheng** [comment]
> @iamxiaojianzheng I used this: https://github.com/newren/git-filter-repo
> 
> I think is rewrites the history completely, so that the email is not in the history at all.
> 
> Sorry for the complications, but maybe you need to etither sync your branch with main, or recreate your fork.

This works better for global changes, but lazygit is the easiest solution I've found so far if you're only changing partial commits. The features are really impressive.

**Pull request closed without merging**